site/tests/RepoCrawler/WebNotificationConsumerTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Tests\RepoCrawler;
+
+use App\Consumer\WebNotificationConsumer;
+use App\Entity\User;
+use App\Util\Notification;
+use Mgilet\NotificationBundle\Manager\NotificationManager;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Tests for the Web Notification Consumer
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+class WebNotificationConsumerTest extends TestCase
+{
+    protected $notification;
+
+    /** @test */
+    public function setUp()
+    {
+        $this->notification = new Notification();
+        $this->notification->setMessage('Your project was added to LibreCores');
+        $this->notification->setSubject('New Project');
+        $this->notification->setType('new_project');
+        $user = new User();
+        $user->setUsername('test');
+        $this->notification->setRecipient($user);
+
+        // Assert if fields are set properly
+        $this->assertEquals('Your project was added to LibreCores', $this->notification->getMessage());
+        $this->assertEquals('New Project', $this->notification->getSubject());
+        $this->assertEquals('new_project', $this->notification->getType());
+        $this->assertEquals('test', $this->notification->getRecipient()->getUsername());
+    }
+
+    /**
+     * @test
+     *
+     * Tests if a notification is added to database
+     */
+    public function test_if_notification_is_added()
+    {
+        $mockLogger = $this->createMock(LoggerInterface::class);
+        $mockNotificationManager = $this->getMockBuilder(NotificationManager::class)
+            ->setMethods(['addNotification'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $webNotificationConsumer = new WebNotificationConsumer($mockLogger, $mockNotificationManager);
+        $webNotificationConsumer->setNotification($this->notification);
+
+        // Test if handle() doesn't requeue notifications in RabbitMQ
+        $this->assertEquals(true, $webNotificationConsumer->handle());
+
+        // Test if notification is added to database
+        $mockNotificationManager->expects($this->once())
+            ->method('addNotification');
+
+        $webNotificationConsumer->handle();
+    }
+}
```

Comment:
The test_if_notification_is_added() test is incomplete, it doesn't explore all possible paths the function can take.
- What happens if shouldHandle() returns true or false? How can you test both cases?
- In which cases returns handle() a 0 or non-zero return value? What can do you to test both cases?

Reply:
@imphil For testing both shouldHandle and handle, we need to set them as public first?

Reply:
Else we need to have a public method that calls them, what's the better option?

Reply:
@imphil I just made the possible tests, please have a look at it